js/eventloop.go

```diff
@@ -0,0 +1,167 @@
+package js
+
+import (
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/dop251/goja"
+	"go.k6.io/k6/js/modules"
+)
+
+// A basic event loop implemnentation, with a cut down setTimeout and
+// handling of unhandled rejected promises
+//
+// A specific thing about this event loop is that it will wait to return
+// not only until the queue is empty but until nothing is reserved.
+// This is in contrast with more common behaviours where it only returns on
+// a specific event/action or when the loop is empty.
+// This is required as in k6 iterations (for which event loop will be primary used)
+// are supposed to be independent and any work started in them needs to finish,
+// but also they need to end when all the instructions are done.
+// Additionally because of this on any error while the event loop will exit it's
+// required to wait on the event loop to be empty before the execution can  continue.
```

Comment:
```suggestion
// eventLoop implements an event loop with a cut-down setTimeout and
// handling of unhandled rejected promises.
//
// A specific thing about this event loop is that it will wait to return
// not only until the queue is empty but until nothing is reserved.
// This is in contrast with more common behaviours where it only returns on
// a specific event/action or when the loop is empty.
// This is required as in k6 iterations (for which event loop will be primary used)
// are supposed to be independent and any work started in them needs to finish,
// but also they need to end when all the instructions are done.
// Additionally because of this on any error while the event loop will exit it's
// required to wait on the event loop to be empty before the execution can continue.
```